lighthouse-viewer/app/src/firebase-auth.js

```diff
@@ -29,52 +29,68 @@ class FirebaseAuth {
       messagingSenderId: '962507201498'
     });
 
-    // Wrap auth state callback in a promise so other parts of the app that
-    // require login can hook into the changes.
-    this.ready = new Promise((resolve, reject) => {
-      firebase.auth().onAuthStateChanged(user => {
-        idb.get('accessToken').then(token => {
-          if (user && token) {
-            this.accessToken = token;
-            this.user = user;
-          }
-          resolve(user);
-        });
-      });
+    /**
+     * Promise which resolves after the first check of auth state. After this,
+     * _accessToken will be set if user logged in and has access token.
+     * @private {!Promise<undefined>}
+     */
+    this._ready = Promise.all([
+      new Promise(resolve => firebase.auth().onAuthStateChanged(resolve)),
+      idbKeyval.get('accessToken')
+    ]).then(([user, token]) => {
+      if (user && token) {
+        this._accessToken = token;
+        this._firebaseUser = user;
+      }
     });
   }
 
+  /**
+   * Returns the GitHub access token if already logged in. If not logged in,
+   * returns null (and will not trigger sign in).
+   * @return {!Promise<?string>}
+   */
+  getAccessTokenIfLoggedIn() {
```

Comment:
I *believe* this is a somewhat more explicit version of the old `ready` promise. Having a firebase `user` also seemed only to be an implementation detail, so it's now entirely internal to the class